release_process.md

```diff
@@ -1,162 +1,475 @@
-- **dry_run**: `false` (use `true` to test first)
-- **draft_release**: `true` (to review release notes before publishing)
-
-The workflow will:
-1. Check for breaking changes automatically
-2. Update all version numbers using bump-my-version
-3. Create a commit with the version update
-4. Create a tag with format `vX.Y.Z`
-5. Push both the commit and tag
-6. Create a GitHub release
-
-The action will automatically generate release notes. **Please review these
-and make any necessary edits.**
-
-Once that release is published, it will trigger publish jobs for Rust, Python, and Java.
-
-## Version Management
-
-### Automated Version Bumping
+# Release Process
+
+Lance uses [semantic versioning](https://semver.org/) and maintains a linear commit history.
+* All pull requests are merged into the `main` branch.
+* Beta releases (or preview releases) are created on-demand from the `main` branch.
+* Stable releases (non-prereleases) are created only after a voting process and come from a release branch `release/vX.Y`. These are typically created once every two weeks.
+* Release Candidates (RC) are created from release branches prior to voting.
+* Patch releases are created by committing fixes directly to the release branch, voting on a new RC, and releasing.
+
+```mermaid
+gitGraph
+    commit
+    branch feature
+    checkout feature
+    commit
+    checkout main
+    merge feature
+    branch bugfix
+    checkout bugfix
+    commit id: "bugfix"
+    checkout main
+    branch "release/v1.4"
+    checkout "release/v1.4"
+    commit tag: "1.4.0-rc.1" tag: "1.4.0"
+    checkout main
+    merge bugfix
+    commit id: "merged"
+    checkout "release/v1.4"
+    cherry-pick id: "merged"
+    commit tag: "1.4.1-rc.1" tag: "1.4.1"
+    checkout main
+    commit tag: "1.5.0-beta.1"
+    
+```
 
-The release process now uses `bump-my-version` configured in `.bumpversion.toml` to:
-- Synchronize versions across all Rust crates
-- Update Python and Java package versions
-- Update all Cargo.lock files automatically
+## Version Semantics
+
+### Version Format
+
+Lance uses semantic versioning with prerelease identifiers:
+- **Stable**: `X.Y.Z` (e.g., `1.3.0`)
+- **Beta**: `X.Y.Z-beta.N` (e.g., `1.3.0-beta.1`, `1.3.0-beta.2`)
+- **RC**: `X.Y.Z-rc.N` (e.g., `1.3.0-rc.1`, `1.3.0-rc.2`)
+
+### Beta Version States
+
+- **beta.0**: Unreleased version (exists on branch but not published)
+  - Created after cutting an RC to mark the next unreleased version
+  - Indicates no preview has been published yet
+- **beta.1+**: Published preview releases
+  - Created when publishing beta preview artifacts
+
+### Publishing Channels
+
+| Language     | Stable release      | RC release                        | Beta release                      |
+|--------------|---------------------|-----------------------------------|-----------------------------------|
+| **Rust**     | crates.io           | Not published (use git tag)       | Not published (use git tag)       |
```

Comment:
I'm thinking of always release them so downstream can just use version instead of playing with git tag.